### PR TITLE
added flag to compress the output files

### LINKED ISSUE #55: Add the possbility to compress the output files with pigz
# nf-core/pangenome feature request

Hi there!

## Describe the solution you'd like

Introduce the `--gzip-output` parameter. Files generated by `seqwish` or `smoothxg` step should be compressed using `pigz`.

### PR DESCRIPTION
**Pull request by AndreaGuarracino**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/pangenome/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/pangenome _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

This fixes #55.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -275,6 +282,20 @@ process odgiDraw {
   """
 }
 
+process gzipOutputFiles {
```
Comment (subwaystation) [inline main.nf:285]: Would it make sense to rename this into `pigzOutputFiles`? To make it clearer, which tool the process is using.

File: main.nf [hunk 2]
```diff
@@ -298,15 +319,15 @@ workflow {
   main:
     if (params.wfmash == false) {
       edyeet(fasta)
-      seqwish(edyeet.out)
+      seqwish(fasta, edyeet.out)
```
Comment (subwaystation) [inline main.nf:322]: Why the additional input for seqwish?

File: nextflow_schema.json [hunk 3]
```diff
@@ -43,6 +43,12 @@
                     "default": true,
                     "hidden": true,
                     "fa_icon": "fas fa-file-csv"
+                },
+                "do_compression": {
+                    "type": "boolean",
+                    "default": false,
+                    "hidden": false,
+                    "fa_icon": "fas fa-file-csv"
```
Comment (subwaystation) [inline nextflow_schema.json:51]: I would look for a different fav? A CSV one does not really fit here.

File: main.nf [hunk 4]
```diff
@@ -298,15 +319,15 @@ workflow {
...
     } else {
       wfmash(fasta)
-      seqwish(wfmash.out)
+      seqwish(fasta, wfmash.out)
```
Comment (subwaystation) [inline main.nf:325]: Did you add this because seqwish usually expects a FASTA and a PAF as input?

Comment (AndreaGuarracino) [reply main.nf:325]: Yes, the FASTA provides the sequences, while the PAF provides information on how they align with each other.

**Review by subwaystation — APPROVED** [review]

**Comment by subwaystation** [comment]
LGTM :)

**Pull request merged**